node/libs/crypto/src/bn254/mod.rs

```diff
@@ -81,6 +81,24 @@ impl PartialEq for SecretKey {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PublicKey(G2);
 
+impl PublicKey {
+    /// Checks if the public key is not the identity element and is in the correct subgroup. Verifying signatures
+    /// against public keys that are not valid is insecure.
+    pub fn is_valid(&self) -> bool {
```

Comment:
it can be private, no?

Reply:
nit: more canonical would be to make it `fn verify(&self) -> Result<()>`, rather than return a bool.

Reply:
I think you should also add a check to SecretKey::decode(), because SecretKey::public() also constructs a PublicKey

Reply:
It can be private, but I think it's better to keep it public. It's a useful method.

Reply:
I mean, all PublicKey objects ever constructed are expected to be valid, so having verify() in the public API will be quite misleading...